examples/calc/CalcTest.fst

```diff
@@ -0,0 +1,111 @@
+module CalcTest
+
+open FStar.Mul
+open FStar.Calc
+open FStar.Preorder
+
```

Comment:
A small, say 2 steps example, that shows the desugaring in the comments would be useful here.

Reply:
Maybe a version of the same program written without any syntactic sugar?